### PR TITLE
Cluster Configuration API

### LINKED ISSUE #331: API for getting list of frequency plan IDs
**Summary:** 

We need an API for getting a list of available frequency plan IDs for a cluster.

**Why do we need this?**

In the console pages for gateway and end device registration, we want to show a dropdown for selecting the frequency plan ID.

**What is already there? What do you see now?**

`pkg/frequencyplans` has the `GetAllIDs` func.

**What is missing? What do you want to see?**

An API that returns the list of available frequency plans. This API needs to be exposed at least on the GS and NS (for the registration of both entities), but considering that it's global config, we can just always expose it if there's a frequency plans fetcher configured.

For the Gateway registration the console will have to call this endpoint on the GS base URL, for EndDevice registration, the console will have to call this endpoint on the NS base URL.

**How do you propose to implement this?**

Define an RPC service for getting "config" from a cluster with a method for getting the list of frequency plans using the `GetAllIDs` method of `pkg/frequencyplans`.

**What can you do yourself and what do you need help with?**

I can do the backend part and let @kschiffer / @bafonins know when it's ready. In the meantime they'll have to work with a simple text field.

### PR DESCRIPTION
**Pull request by htdvisser**
**Summary:**

This PR closes #331 by introducing a `Configuration` service in our API. This service will be extendend in the future, right now I only added the `ListFrequencyPlans` RPC.

cc: @kschiffer and @bafonins (see the http call below)

**Changes:**

- Define the protos and recompile
- Implement the RPC
- Small change to the frequencyplans package to make frequency plans with a base ID take the base frequency from their base.

**Notes for Reviewers:**

Test with `go run ./cmd/ttn-lw-cli ed list-frequency-plans --base-frequency 868` or with `curl "localhost:1885/api/v3/configuration/frequency-plans?base_frequency=868"`. The `base_frequency` is optional.

**Release Notes: _(optional)_**

- Add the `Configuration` gRPC service with the `ListFrequencyPlans` RPC

### CODE REVIEW

**Comment by kschiffer** [comment]
Updated SDK api definitions as well.

File: api/configuration_services.proto [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import "github.com/lyft/protoc-gen-validate/validate/validate.proto";
+import "google/api/annotations.proto";
+
+package ttn.lorawan.v3;
+
+option go_package = "go.thethings.network/lorawan-stack/pkg/ttnpb";
+
+message ListFrequencyPlansRequest {
+  // Optional base frequency in MHz for hardware support (433, 470, 868 or 915)
+  uint32 base_frequency = 1;
```
Comment (rvolosatovs) [inline api/configuration_services.proto]: The comment implies that there are only 4 possible values, why is this field not validated?

File: api/configuration_services.proto [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+message FrequencyPlanDescription {
+  string id = 1 [(gogoproto.customname) = "ID"];
+  // The ID of the frequency that the current frequency plan is based on.
+  string base_id = 2 [(gogoproto.customname) = "BaseID"];
+  string name = 3;
+  // Base frequency in MHz for hardware support (433, 470, 868 or 915)
+  uint32 base_frequency = 4;
```
Comment (rvolosatovs) [inline api/configuration_services.proto]: IDEM

File: api/configuration_services.proto (hunk 1 above)
Comment (johanstokking) [reply api/configuration_services.proto]: These are examples and are subject to change. No reason for API validation here.

Comment (rvolosatovs) [reply api/configuration_services.proto]: The comment should mention `e.g` or `for example` then.

**Pull request merged**